console/src/main/java/org/georchestra/console/ws/backoffice/users/UserInfoExporterImpl.java

```diff
@@ -0,0 +1,273 @@
+        try {
+            SearchExecutor executor = new SearchExecutor();
+            executor.setBaseDn(this.userSearchBaseDn);
+            log.info("Running search '{}'", filter);
+            result = executor.search(cf, filter).getResult();
+            log.info("User query successful: {}", result);
+        } catch (LdapException e) {
+            log.error("Error running {}", filter, e);
+            throw new IllegalStateException(e);
+        }
+
+        StringWriter writer = new StringWriter();
+        try {
+            new LdifWriter(writer).write(result);
+        } catch (IOException e) {
+            log.error("Error generating LDIF file", e);
+            throw new IllegalStateException(e);
+        }
+        String ldifContents = writer.toString();
+        log.info("Returning LDIF: {}", ldifContents);
+        return ldifContents;
+    }
+
+    @Override
+    public @NonNull String toVcf(@NonNull Account account) {
+        VCard v = new VCard();
+        FormattedName f = new FormattedName(account.getGivenName() + " " + account.getSurname());
+        v.addFormattedName(f);
+        v.addEmail(account.getEmail(), EmailType.WORK);
+        v.addTelephoneNumber(account.getPhone(), TelephoneType.WORK);
+        v.addTitle(account.getTitle());
+        Address a = new Address();
+        a.setPostalCode(account.getPostalCode());
+        a.setStreetAddress(account.getPostalAddress());
+        a.setPoBox(account.getPostOfficeBox());
+        a.setLocality(account.getLocality());
+        v.addAddress(a);
+        v.addTelephoneNumber(account.getMobile(), TelephoneType.CELL);
+
+        return v.write();
+    }
+
+    @Override
+    public @NonNull String exportUsersAsCsv(@NonNull String... userNames) throws DataServiceException {
+        StringBuilder res = new StringBuilder();
+        res.append(OUTLOOK_CSV_HEADER); // add csv outlook header
+        for (String user : userNames) {
+            try {
+                Account a = accountDao.findByUID(user);
+                res.append(toCsv(a));
+            } catch (NameNotFoundException e) {
+                log.error("User [{}] not found, skipping", user, e);
+            }
+        }
+
+        return res.toString();
+    }
+
+    @Override
+    public @NonNull String exportUsersAsVcard(@NonNull String... users) throws Exception {
+        StringBuilder ret = new StringBuilder();
+        for (String user : users) {
+            try {
+                Account a = accountDao.findByUID(user);
+                ret.append(toVcf(a));
+            } catch (NameNotFoundException e) {
+                log.error("User [{}] not found, skipping", user, e);
+            }
+        }
+
+        return ret.toString();
+    }
+
+    @Override
+    public @NonNull String toCsv(@NonNull Account account) {
+
+        StringBuilder csv = new StringBuilder();
+
+        csv.append(toFormatedString(account.getCommonName()));
```

Comment:
Here is according to csv header a first name expected, not concatenation of firstname and lastname !

Reply:
I don't know, that method is taken from `AccountImpl.toCsv()`, which I think is misplaced in `AccountImpl`. Why would an `Acccount` need to know how to export itself to any format (there's AccountImpl.toCsv() and AccountImpl.toVcf(), which is weird. Idea here was to move that logic to UserInfoExport and remove it from Account)

Reply:
Finished the migration from Account to UserInfoExporter as per 35fb120dba3a710be0e4500c673bea8a8df94b86. Resolving this conversation as it's outside the scope of this PR (unrelated to GDPR and existing, old code)